### PR TITLE
Fix destination marker not drawn on Style reload

### PR DESCRIPTION
**Pull request by danesfeder**
## Description

The `NavigationMapboxMap` is not redrawing the `Symbol` for the destination marker on Style reload.

- Fixes https://github.com/mapbox/mapbox-navigation-android/issues/1778

## What's the goal?

Ensure we maintain the current state of the `MapboxMap` between style loads. 

## How is it being implemented?

Adding a `SymbolOnStyleLoadedListener` that holds the `NavigationSymbolManager` and redraws based on the current data in the manager.  

## Screenshots or Gifs

**Before**

ezgif com-video-to-gif

**After**

ezgif com-video-to-gif

## How has this been tested?

Can be tested with the changes here in `ComponentNavigationActivity`:
- Long press to load / draw a route
- Hit "start navigation" icon which is currently rigged to reload the `Style` to `DARK`

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by LukasPaczos** [comment]
The `SymbolManager` has a feature of keeping annotations alive across style changes. The issue here is, that when the style changes, you're not re-adding the icon image of the marker, so the manager doesn't have a resource to reference anymore.

The gist is, that you need to call something like
```
mapView.addOnDidFinishLoadingStyleListener(new MapView.OnDidFinishLoadingStyleListener() {
  @Override
  public void onDidFinishLoadingStyle() {
    mapboxMap.getStyle(new Style.OnStyleLoaded() {
      @Override
      public void onStyleLoaded(@NonNull Style style) {
        Bitmap markerBitmap = ThemeSwitcher.retrieveThemeMapMarker(mapView.getContext());
        style.addImage(MAPBOX_NAVIGATION_MARKER_NAME, markerBitmap);
      }
    });
  }
});
```
wherever it's best based on the navigation setup. The plugin will take care of the rest.

**Comment by danesfeder** [comment]
@LukasPaczos great catch, I updated the code here to re-add the marker bitmap after the new style is loaded and it works great - thank you!

**Comment by Guardiola31337** [comment]
@LukasPaczos 
> he issue here is, that when the style changes, you're not re-adding the icon image of the marker, so the manager doesn't have a resource to reference anymore.

> The plugin will take care of the rest.

Quick question, would it make sense for the `SymbolManager` to keep track of this as well?

**Comment by LukasPaczos** [comment]
@Guardiola31337 adding an image is a separate entry point to the style object and exposing a duplicate method on the manager's side just to track additions would cause confusion imo. But maybe we could rethink that as an options parameter?. Would you mind cutting a ticket in the plugins repo?

**Comment by Guardiola31337** [comment]
@LukasPaczos 
> Would you mind cutting a ticket in the plugins repo?

Sure thing! Done 👉 https://github.com/mapbox/mapbox-plugins-android/issues/860

**Comment by danesfeder** [comment]
@Guardiola31337 thanks for ticketing that, is this okay to ✅ then?

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java [hunk 1]
```diff
@@ -582,6 +582,8 @@ private void initializeNavigationSymbolManager(MapView mapView, MapboxMap mapbox
     mapboxMap.getStyle().addImage(MAPBOX_NAVIGATION_MARKER_NAME, markerBitmap);
     SymbolManager symbolManager = new SymbolManager(mapView, mapboxMap, mapboxMap.getStyle());
     navigationSymbolManager = new NavigationSymbolManager(symbolManager);
+    SymbolOnStyleLoadedListener onStyleLoadedListener = new SymbolOnStyleLoadedListener(mapboxMap, markerBitmap);
+    mapView.addOnDidFinishLoadingStyleListener(onStyleLoadedListener);
```
Comment (Guardiola31337) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java]: Quick question @LukasPaczos I've noticed that we're not removing / releasing the listener from our side. @danesfeder mentioned that this wasn't needed. Is `MapView#onDestroy` releasing all the resources / making sure that any of the listeners don't leak? If so, is this something remarked in the docs? I only wanted to make sure that customers using the APIs don't mess up / the docs reflect how to properly use them. As a developer if I add a listener I'd normally try removing it afterwards based on the lifecycle implemented from "my" side so I think we should make this 💯 clear.

Comment (LukasPaczos) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java]: > Is MapView#onDestroy releasing all the resources / making sure that any of the listeners don't leak?

Yes :)

Comment (Guardiola31337) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java]: Awesome thanks for confirming! I believe that removing it explicitly isn't harmful, right?

Comment (LukasPaczos) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java]: Shouldn't be harmful in any way.

Comment (Guardiola31337) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java]: Yeah, makes sense! As mentioned, I think it's a good practice to remove all the resources previously added. But yeah thanks for confirming that we're backing up if any developers forget, so we don't crash 💪

**Pull request merged**